### DISCUSSION TITLE
How can you generate 'properties' entry, even for Classes without fields?

### DISCUSSION THREAD

**Discussion by sashirestela** [victools/jsonschema-generator#433] [Q&A] [answered]
Is there any configuration parameter to generate the `properties` entry, even for Classes without fields?

For example, for the following class:
```java
class EmptyClass {
}
```
Currently, I'm getting this:
```json
{
    "type":"object"
}
```
But I want to get this one:

```json
{
    "type":"object",
    "properties":{}
}
```
This is my Java code:
```java
var jacksonModule = new JacksonModule(JacksonOption.RESPECT_JSONPROPERTY_REQUIRED);
var configBuilder = new SchemaGeneratorConfigBuilder(SchemaVersion.DRAFT_2020_12,
        OptionPreset.PLAIN_JSON)
        .with(jacksonModule)
        .without(Option.SCHEMA_VERSION_INDICATOR);
var config = configBuilder.build();
var generator = new SchemaGenerator(config);
var jsonSchema = generator.generateSchema(myClass);
```

**Comment by CarstenWickner** [comment, marked as answer]
Hi @sashirestela,

While there is no standard `Option` for it or anything, you can achieve that of course:
```java
// add the properties keyword to all empty object definitions
configBuilder.forTypesInGeneral().withTypeAttributeOverride((node, scope, context) -> {
    JsonNode typeValue = node.get(context.getKeyword(SchemaKeyword.TAG_TYPE));
    String objectTypeValue = context.getKeyword(SchemaKeyword.TAG_TYPE_OBJECT);
    if (typeValue == null || node.has(context.getKeyword(SchemaKeyword.TAG_PROPERTIES))) {
        // nothing to do here
    } else if (typeValue.isTextual() && objectTypeValue.equals(typeValue.textValue())) {
        node.putObject(context.getKeyword(SchemaKeyword.TAG_PROPERTIES));
    } else if (typeValue.isArray()) {
        for (JsonNode singleTypeValue : typeValue) {
            if (objectTypeValue.equals(singleTypeValue.textValue())) {
                node.putObject(context.getKeyword(SchemaKeyword.TAG_PROPERTIES));
                return;
            }
        }
    }
});
```

That being said: why would you want this?
It's not changing the schema in any meaningful way. An empty `"properties"` node is redundant/inconsequential, isn't it?

**Reply by sashirestela** [reply]
Thanks for your answer @CarstenWickner. From a Json-schema point of view you are right, however, in my context I need empty properties. I'm working in a Java integration with OpenaAI called [simple-openai](https://github.com/sashirestela/simple-openai) and I'm using classes to describe functions which are going to be used in chats and OpenAI is expecting an empty value when functions does not have parameters (or the same when the class describing the function does not have fields):

https://community.openai.com/t/parameters-property-isnt-optional-as-theyve-mentioned-in-doc/265256
